packages/x-data-grid/package.json

```diff
@@ -9,7 +9,6 @@
     "url": "https://github.com/mui/mui-x/issues"
   },
   "homepage": "https://mui.com/x/react-data-grid/",
-  "sideEffects": false,
```

Comment:
Importing the CSS stylesheet directly requires this flag to be removed. I'm still wondering if it would be better to let users import our stylesheet directly instead of doing it in our codebase. It would let us keep this flag, and it would let users control when the stylesheet gets imported.

Reply:
This is because of tree-shaking, correct?
What if instead a single file we had a CSS file per component? If the js file is included, then the css file will be included too, and we can keep `sideEffects: false`

Reply:
What's preventing you from updating your babel plugin to emit a css file for each js file instead of rolling it up to a single stylesheet? It would simplify everything and fix the sheets from being tree shaken away by this flag.

Reply:
Oh, we can mark specific files in `sideEffects`: https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

Reply:
cc @brijeshb42

Reply:
This what I have done with `mui-next` - https://github.com/mui/mui-next/pull/73
Adding `sideEffects: false` works with it even with barrel file imports and only the actually used component's css files were imported.

Reply:
Yes, it's because of tree-shaking. How does splitting the CSS file into one per component affect this flag though? IIUC, the stylesheet import is a side-effect whether it's a single file or multiple files.

Reply:
> the stylesheet import is a side-effect whether it's a single file or multiple files.

If a JS file imports a CSS file that only impacts the code in that JS file, then tree shaking shouldn't cause any issues, right?

Reply:
What I observed is that the bundler removed the `import './index.css'` statement because an import without side-effects should be removable at any time, whether it's a single-stylesheet at the root or an `import './SomeSpecificComponent.css'` in a specific component. In my understanding, this issue is present for single or multiple files.

Note that I'm not against splitting the CSS in multiple files, but I don't see how it would fix the `sideEffects: false` behavior.

Reply:
> What I observed is that the bundler removed the import './index.css' statement because an import without side-effects should be removable at any time

You're correct, here's what Webpack docs say about this:

> Note that any imported file is subject to tree shaking. This means if you use something like css-loader in your project and import a CSS file, it needs to be added to the side effect list so it will not be unintentionally dropped in production mode:
Source: https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

In our case, I think this should do the job:
```json
"sideEffects": ["**/*.css"]
```